### PR TITLE
fix(ButtonMenu): refactor 

### PR DESCRIPTION
**Pull request by qbi11y**
Contributes to #2076 

Refactored button menu, now uses button set instead of overflow menu. Verified variations with design spec.

### CODE REVIEW

File: packages/cloud-cognitive/src/components/ButtonMenu/_button-menu.scss [hunk 1]
```diff
@@ -18,17 +19,106 @@
...
+      }
+    }
+  }
+  .#{$block-class} .#{$carbon-prefix}--btn--lg:not(.bx--btn--icon-only) {
+    min-height: $spacing-09;
+    align-items: center;
+    padding-top: $spacing-04;
+  }
+
+  .#{$block-class}__sm .#{$carbon-prefix}--btn,
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/ButtonMenu/_button-menu.scss:101]: Can you remove these styles? Those must be left over from the nested menu prototype.

File: packages/cloud-cognitive/src/components/ButtonMenu/_button-menu.scss [hunk 2]
```diff
@@ -18,17 +19,106 @@
...
+        background-color: $hover-ui;
+      }
+
+      // Remove visual divider if next sibling is not a button menu item button
+      &:has(+ :not(.#{$block-class}__item-button))::before {
+        background-color: transparent;
+      }
+    }
+
+    .#{$carbon-prefix}--accordion
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/ButtonMenu/_button-menu.scss:78]: Same comment as [here](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/2442/files#r1017076898)

**Review by elycheea — CHANGES_REQUESTED** [review]

**Comment by elycheea** [comment]
I mentioned this over Slack but we’ll also need to look into how we can get ButtonMenu and ComboButton working more consistently. Because ButtonMenu currently uses ButtonSet, navigating the menu is different than ComboButton although they look visually similar.

**Comment by qbi11y** [comment]
I commented out some code causing issues because I didn't understand what was needed to resolve. I updated the PR so the team can see everything and offer insight. The test for ButtonWithOverFlow fails if I uncomment the section that mimics the user click. I know it mimics a click and is expecting a function to be called. All that seemed to be in place so not sure why it wasn't passing.

File: packages/cloud-cognitive/src/components/ButtonMenu/ButtonMenu.js [hunk 3]
```diff
@@ -24,70 +26,83 @@ const componentName = 'ButtonMenu';
 
 // Default values for props
 const defaults = {
-  size: 'default',
+  size: 'lg',
   kind: 'primary',
+  open: false,
+  isDanger: false,
+  isDisabled: false,
```
[the remark this answers is not in the document]
Comment (qbi11y) [reply packages/cloud-cognitive/src/components/ButtonMenu/ButtonMenu.js:33]: I have started my storybook research :). I was working with the wrong assumption. Reading the documentation is helping me a bit. I can remove these as suggested.

**Comment by elycheea** [comment]
After we fix and merge #2442, we can move ButtonMenu to canary + Storybook fixes.

**Pull request closed without merging**